strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/remote/heartbeat/RemoteRepositoryAlivenessCacheManagerTest.java

```diff
@@ -0,0 +1,140 @@
+package org.carlspring.strongbox.storage.repository.remote.heartbeat;
+
+import org.carlspring.strongbox.config.*;
+import org.carlspring.strongbox.config.hazelcast.HazelcastConfiguration;
+import org.carlspring.strongbox.config.hazelcast.HazelcastInstanceId;
+import org.carlspring.strongbox.data.CacheManagerTestExecutionListener;
+import org.carlspring.strongbox.data.CacheName;
+import org.carlspring.strongbox.storage.repository.remote.RemoteRepository;
+import org.carlspring.strongbox.testing.NullLayoutConfiguration;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.*;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestExecutionListeners;
+
+import javax.inject.Inject;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+@ActiveProfiles({ "test",
+                  "RemoteRepositoryAlivenessCacheManagerTestConfig" })
+@TestExecutionListeners(listeners = { CacheManagerTestExecutionListener.class },
+                        mergeMode = TestExecutionListeners.MergeMode.MERGE_WITH_DEFAULTS)
+public class RemoteRepositoryAlivenessCacheManagerTest
```

Comment:
`strongbox-storage-api` tests are configured to be run `concurrently` (see `src/test/resources/junit-platform.properties`). You should make sure every method uses its own `remoteRepository` or you should mark the test as `@Execution(SAME_THREAD)`. Otherwise the results are unpredictable.

Reply:
Hi, thanks for your feedback. I have inlined the instances of `remoteRepository`, please have a look at the latest commits.

Reply:
it seems we better use `@Execution(SAME_THREAD)` here because test methods will use same cache instance anyway.

Reply:
Ah, nice catch. I agree, I'll revert the previous fix and add `@Execution(SAME_THREAD)` instead.